### DISCUSSION TITLE
HTTPS workaround

### DISCUSSION THREAD

**Discussion by CasaPas** [kiwix/.github#8] [General]
I just installed Kiwix in a docker container on my Unraid server.

Some of the Zim files won't open due to the known "HTTPS service worker" issue.

The problem being, that generally you access Kiwix via your remote browser pointing to the IP address of the Unraid server, rather than "localhost" if the files were on the host machine.

I managed to get these Zim files working, by installing a Firefox docker on the Unraid server, and setting its networking to "host" rather than "bridge". Basically this browser has access to the "host" or local files. Using this docker container browser, you are now able to access "localhost:8040" version of Kiwix.

I can now connect to my Kiwix server from a remote machine, looking at the Unraid Firefox browser using localhost:8040/ as an address inside the Firefox docker rather than using the IP address of the Unraid server in the remote or Unraid browser.

It's a bit cumbersome, but it works until the release of the new version which is due to solve this issue.

So;

1: Local browser -> 2: Unraid Browser (IPofUnraidServer:3000) -> 3: localhost:8040

1: Tab in my remote machine pointing to the Unraid docker gui for Firefox (X.X.X.X:3000)
2: Tab inside the Firefox docker pointing to the "local" web gui for Kiwix (localhost:8040)
3: Kiwix running "locally" on the host Unraid system.

Clear as mud, as we say in Ireland.

Hope that helps someone.

**Comment by rgaudin** [comment]
Thank you @CasaPas for the workaround.
I am pleased to inform you that we removed all ZIMs requiring ServiceWorker from the [download server/library](https://download.kiwix.org/zim/zimit/) today.

Files listed there does not require SW. Content that were previously there and not listed yet are most likely to-be-rebuilt. A couple content have issues that are being tracked though.

**Comment by CasaPas** [comment]
No problem, thank you for your efforts with this project.

Murphy's law, I just finished downloading most of the ZIM's. Seems like I have a new weekend project!!
